xbmc/pvr/channels/PVRChannelGroups.cpp

```diff
@@ -212,15 +208,32 @@ void CPVRChannelGroups::SortGroupsByLocalOrder()
 
 void CPVRChannelGroups::SortGroups()
 {
-  const bool backendOrderSort =
-      m_settings.GetBoolValue(CSettings::SETTING_PVRMANAGER_BACKENDCHANNELGROUPSORDER);
+  bool orderChanged{false};
+  {
+    std::unique_lock<CCriticalSection> lock(m_critSection);
 
-  if (backendOrderSort)
-    SortGroupsByBackendOrder();
-  else
-    SortGroupsByLocalOrder();
+    const bool backendOrderSort{
+        m_settings.GetBoolValue(CSettings::SETTING_PVRMANAGER_BACKENDCHANNELGROUPSORDER)};
 
-  CServiceBroker::GetPVRManager().PublishEvent(PVREvent::ChannelGroupsInvalidated);
+    const std::vector<std::shared_ptr<CPVRChannelGroup>> groupsInOldOrder{m_groups};
+
+    if (backendOrderSort)
+      SortGroupsByBackendOrder();
+    else
+      SortGroupsByLocalOrder();
+
+    for (size_t i = 0; i < m_groups.size(); ++i)
+    {
+      // Note: Index operator preferred over at(i) here. No out-of-bounds checks needed.
+      if (m_groups[i]->GetPath() != groupsInOldOrder[i]->GetPath())
```

Comment:
Is it necessary to compare the path? As the sorting only shuffles around pointers I'd assume that comparing the pointers should also be sufficient. In that case we could just do:
```cpp
orderChanged = m_groups != groupsInOldOrder;
```